crates/ruff_python_trivia/src/tokenizer.rs

```diff
@@ -182,7 +182,7 @@ fn to_keyword_or_other(source: &str) -> SimpleTokenKind {
         "case" => SimpleTokenKind::Case,
         "with" => SimpleTokenKind::With,
         "yield" => SimpleTokenKind::Yield,
-        _ => SimpleTokenKind::Other, // Potentially an identifier, but only if it isn't a string prefix. We can ignore this for now https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
+        _ => SimpleTokenKind::Name, // Potentially an identifier, but only if it isn't a string prefix. The caller is responsible for enforcing that constraint.
```

Comment:
I don't feel comfortable with this. It violates the constraint outlined in the `SimpleTokenizer` lexer where it guarantees to generate valid results for as long as you don't point it in the middle of a token, string, or comment.

It's also not a constraint easily guarded against because it yields incorrect results whenever you start lexing at the start of ANY expression. 

I recommend exploring how expensive it is to handle string prefixes.

Reply:
I’m confused - this is an internal method? The SimpleTokenizer handles this internally.

Reply:
We do properly handle string prefixes in the simple tokenizer - see below (and the test cases). In what case would this yield incorrect results?

Reply:
I see now. You already do what I recommended and I misunderstood the comment. I assumed that the "caller" referred to in the comment is the caller of `SimpleTokenizer::next`.

Reply:
I clarified in the comment -- sorry!